### PR TITLE
Add selection to table and allow deleting selected rows

### PR DESCRIPTION
**Pull request by allysonjp715**
deletegif

### CODE REVIEW

**Comment by allysonjp715** [comment]
[Bulk selection/edit/deletion](https://trello.com/c/5lZN61SP/277-bulk-selection-edit-deletion)

File: verification/curator-service/ui/src/components/LinelistTable.tsx [hunk 1]
```diff
@@ -314,55 +318,92 @@ class LinelistTable extends React.Component<Props, LinelistTableState> {
...
+                                              <DeleteIcon />
+                                          </span>
+                                      ),
+                                      tooltip: 'Delete selected rows',
+                                      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                                      onClick: (_: any, rows: any): void => {
+                                          rows.forEach((row: TableRow) =>
+                                              this.deleteCase(row),
+                                          );
+                                          this.tableRef.current.onQueryChange();
```
Comment (axmb) [inline verification/curator-service/ui/src/components/LinelistTable.tsx:372]: Is it something about doing this within `actions` that necessitates this update?

For instance, with the `editable.onRowDelete` item, the table automatically refreshes based on the deletion.

Comment (allysonjp715) [reply verification/curator-service/ui/src/components/LinelistTable.tsx:372]: Yea the editable actions do that automatically but these actions do not.

File: verification/curator-service/ui/src/components/LinelistTable.tsx [hunk 2]
```diff
@@ -314,55 +318,92 @@ class LinelistTable extends React.Component<Props, LinelistTableState> {
...
+                                          <span aria-label="details">
+                                              <VisibilityIcon />
+                                          </span>
+                                      ),
+                                      onClick: (e, row): void => {
+                                          // Somehow the templating system doesn't think row has an id property but it has.
+                                          const id = (row as TableRow).id;
+                                          history.push(`/cases/view/${id}`);
+                                      },
+                                      tooltip: 'View this case details',
```
[the remark this answers is not in the document]
Comment (allysonjp715) [reply verification/curator-service/ui/src/components/LinelistTable.tsx:402]: Yes I tried to keep the concat in, but [this bug](https://github.com/mbrn/material-table/issues/676) prevented it. It's a very strange bug and I can't really seem to understand when it is triggered and when it isn't. For example I just tried to pull the details action out into a constant and the bug happened and the type system wouldn't allow it :(

**Pull request merged**